### ISSUE TITLE
Destination service behaves incorrectly with Host network IPs

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#4331]
While testing #4274, I ran into a surprising issue.  This [manifest](https://gist.github.com/olix0r/8c829f330eed7ad08d0a9552e8b97f38) configures a prometheus instance as well as a _node-exporter_ pod (which runs on the Host network).

Initially, the proxy discovers the endpoint via the destination service:
```
[    30.975459286s] DEBUG outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: linkerd2_proxy_api_resolve::resolve: Add endpoints=1
```

Some time later, however, we get an event that removes the endpoint, which removes the endpoint from the balancer:
```
[    64.131592841s]  INFO outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: linkerd2_proxy_api_resolve::resolve: No endpoints
[    64.131913415s] DEBUG outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: tower_balance::p2c::service: updating from discover
[    64.131923668s] TRACE outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: tower_balance::p2c::service: remove
[    64.131942324s] TRACE outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: tower_balance::p2c::service: poll_unready ready=0 pending=0
[    64.131955902s] TRACE outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: tower_balance::p2c::service: ready service became unavailable
[    64.131971227s] TRACE outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: linkerd2_timeout::failfast: Pending
[    64.131983824s] TRACE outbound:accept{peer.addr=10.42.0.77:53162}:source{target.addr=172.18.0.2:9100}:logical{addr=172.18.0.2:9100}:balance{addr=172.18.0.2:9100}: linkerd2_buffer::dispatch: Waiting for inner service
```

Why is the no endpoint update sent by the controller?

This behavior has not been observed with any other endpoints in the test configuration (which are pod-network endpoints). This only seems to be observed with Host-network endpoints.

**Comment by adleong** [comment]
This is probably related to our special treatment of pods with host networking.  For context, see the discussion which starts at https://github.com/linkerd/linkerd2/pull/3595#pullrequestreview-308643978

TLDR is that we omit pods with host networking because for a single host IP, there may be multiple pods which use that IP so we don't know which pod metadata to use.

Probably the correct solution here is to simply return an address back without any pod metadata for host IPs instead of omitting them entirely.

**Comment by adleong** [comment]
(though I can't explain why this query returns an endpoint in the first place)

**Comment by olix0r** [comment]
Anecdotally, @siggy and I just saw what looks like a similar issue with an endpoint on the pod-network:

```
[  4085.190275879s]  INFO outbound:accept{peer.addr=10.244.11.60:37276}:source{target.addr=10.244.10.36:4191}:logical{addr=10.244.10.36:4191}:balance{addr=10.244.10.36:4191}: linkerd2_proxy_api_resolve::resolve: No endpoints
```

So the initial description may not be correct.
